source/extensions/tracers/skywalking/tracer.h

```diff
@@ -51,16 +52,27 @@ using TracerPtr = std::unique_ptr<Tracer>;
 
 class Span : public Tracing::Span {
 public:
-  Span(absl::string_view name, TracingContextPtr tracing_context, Tracer& parent_tracer)
+  Span(absl::string_view name, absl::string_view protocol, TracingContextPtr tracing_context,
+       Tracer& parent_tracer)
       : parent_tracer_(parent_tracer), tracing_context_(tracing_context),
         span_entity_(tracing_context_->createEntrySpan()) {
     span_entity_->startSpan({name.data(), name.size()});
+    skywalking::v3::SpanLayer layer;
+    if (absl::StrContains(protocol, "HTTP")) {
+      // TraceContext.protocol of http is parsed from http message, which value could be HTTP/1.1,
+      // etc.
+      layer = skywalking::v3::SpanLayer::Http;
+    } else {
+      skywalking::v3::SpanLayer_Parse(std::string(protocol), &layer);
+    }
+    span_entity_->setSpanLayer(layer);
   }
-  Span(absl::string_view name, Span& parent_span, TracingContextPtr tracing_context,
-       Tracer& parent_tracer)
+  Span(absl::string_view name, const skywalking::v3::SpanLayer& span_layer, Span& parent_span,
```

Comment:
nit: const ref is unnecessary here for simple enum.

Reply:
I removed const ref, and added test cases, please have a look.